test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestApiParser.java

```diff
@@ -108,13 +107,6 @@ public ClientYamlSuiteRestApi parse(String location, XContentParser parser) thro
                                 restApi.addParam(param, PARAMETER_PARSER.parse(parser, null).isRequired());
                             }
                         }
-
-                        if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
-                            innerLevel++;
-                        }
-                        if (parser.currentToken() == XContentParser.Token.END_OBJECT) {
-                            innerLevel--;
-                        }
```

Comment:
what happens with this fix if you have something like:

```
    "url" : {
        "parts" : {
        },
        "path" : "my_path"
    }
```

Does the path get parsed?

Reply:
I am not sure that this behaviour is correct though, so I would fix that and add a test that makes sure the behaviour for this scenario is correct. Do you see what I mean?

Reply:
>I am not sure that this behaviour is correct though

@javanna What you mean by *this behaviour*?
Currently the following api spec is parsed without a problem.
```
{
  "test": {
    "documentation": "whatever",
    "methods": [ "GET", "POST" ],
    "url": {
      "parts": {},
      "path": "/my_path"
    },
    "body": {
      "description" : "whatever",
      "required" : true
    }
  }
}
```
Or do you mean that the `path` needs to be validated as a string, although it will not be stored in the api spec.

TBH I find the usage of `innerLevel` (especially when combined with object parser) confusing and, as raised by the issue, unreliable and order dependent.

Reply:
You are right, I forgot that `path` is ignored anyways, and I misread the parsing code thinking that the inner parsers would not consume the inner end object, and then we could end up exiting the loop before reading the whole container object. All good , I think your change is good, I will trigger tests.